backend/s3/s3.go

```diff
@@ -2972,6 +2978,39 @@ func (o *Object) Open(ctx context.Context, options ...fs.OpenOption) (in io.Read
 	if err != nil {
 		return nil, err
 	}
+
+	var size int64
```

Comment:
Can you factor this code out of readMetaData rather than duplicating it?

So make a function called 

    func (o *Object) readMetaDataFromResp(ctx context.Context, resp *http.Response) (err error) {

And factor that code out of `readMetaData` and call it here.

Reply:
resp objects are not http.Response. So, it's difficult for me because the two resp objects are different types of s3.HeadObjectOutput and s3.GetObjectOutput.

Reply:
You can read what you need out of the object first and call a factored out method with a few more argument?

Reply:
I removed duplicated code and made a new function of setMetaData(...).